java/apirag-groq/src/main/java/com/datasqrl/ai/models/groq/GroqTokenCounter.java

```diff
@@ -34,7 +34,7 @@ public int countTokens(FunctionDefinition function) {
     return countTokens(jsonString);
   }
 
-  public static GroqTokenCounter of(GroqChatModel model) {
-    return new GroqTokenCounter(HuggingFaceTokenizer.newInstance(model.tokenizerName));
+  public static GroqTokenCounter of(GroqModelConfiguration model) {
```

Comment:
Why go through the configuration, since we have a `GroqModelType` here? Is this on purpose, so that we always interact with the Configuration instead of the ModelType?